doc/tutorials/reinforcement_learning/reinforcement_learning.txt

```diff
@@ -0,0 +1,239 @@
+/*!
+@file rl.txt
+@author Sriram S K
+@brief Tutorial for how to use the Reinforcement Learning module in mlpack.
+
+@page rltutorial Reinforcement Learning Tutorial
+
+@section intro_rltut Introduction
+
+Reinforcement Learning is one of the hottest topics right now, with 
+interest surging after DeepMind published their article on training 
+deep neural networks to play Atari games to great success. mlpack
+implements a complete end-to-end framework for Reinforcement Learning,
+featuring multiple environments, policies and methods. Of course, 
+custom environments and policies can be used and plugged into the
+existing framework with no runtime overhead.
+
+mlpack implements typical benchmark environments (Acrobot, Mountain car etc.),
+commonly used policies and replay methods and supports asynchronous
+learning as well. In addition, it can [communicate](https://github.com/zoq/gym_tcp_api) 
+with the OpenAI Gym toolkit for more environments.
+
+@section toc_rltut Table of Contents
+
+This tutorial is split into the following sections:
+
+ - \ref intro_rltut
+ - \ref toc_rltut
+ - \ref environment_rltut
+ - \ref agent_components_rltut
+ - \ref q_learning_rltut
+ - \ref further_rltut
+
+@section environment_rltut Reinforcement Learning Environments
+
+mlpack implements a number of the most popular environments used for testing
+RL agents and algorithms. These include the Cart Pole, Acrobot, Mountain Car
+and their variations. Of course, as mentioned above, you can communicate with 
+OpenAI Gym for other environments, like the Atari video games.
+
+A key component of mlpack is its extensibility. It is a simple process to create
+your own custom environments, specific to your needs, and use it with mlpack's 
+RL framework. All the environments implement a few specific methods and classes 
+which are used by the agents while learning. 
+
+- \c State: The State class is a representation of the environment. For the CartPole,
```

Comment:
That does sound like a good idea, but perhaps it might be a little too detailed for a tutorial? In particular, the Sample method, which does a number of other things which would in turn require explanations.

Reply:
I see your point, I guess it will introduce more confusion.